l10n.py

```diff
@@ -48,235 +63,325 @@
-    # script is a capitalized 4 alpha ISO 15924 code and region is an uppercase 2 alpha ISO 3166 code
-    def preferredLanguages(self):
+        with suppress(ValueError):
+            return locale.atoi(string)
+
+        with suppress(ValueError):
+            return locale.atof(string)
 
-        if platform=='darwin':
+        return None
+
+    def preferred_languages(self) -> Iterable[str]:
+        """
+        Return a list of preferred language codes.
+
+        Returned data is in RFC4646 format (i.e. "lang[-script][-region]")
+        Where lang is a lowercase 2 alpha ISO 693-1 or 3 alpha ISO 693-2 code
+        Where script is a capitalized 4 alpha ISO 15924 code
+        Where region is an uppercase 2 alpha ISO 3166 code
+
+        :return: The preferred language list
+        """
+        if platform == 'darwin':
             return NSLocale.preferredLanguages()
 
-        elif platform=='win32':
-
-            def wszarray_to_list(array):
-                offset = 0
-                while offset < len(array):
-                    sz = ctypes.wstring_at(ctypes.addressof(array) + offset*2)
-                    if sz:
-                        yield sz
-                        offset += len(sz)+1
-                    else:
-                        break
-
-            num = ctypes.c_ulong()
-            size = ctypes.c_ulong(0)
-            if (GetUserPreferredUILanguages(MUI_LANGUAGE_NAME, ctypes.byref(num), None, ctypes.byref(size)) and size.value):
-                buf = ctypes.create_unicode_buffer(size.value)
-                if GetUserPreferredUILanguages(MUI_LANGUAGE_NAME, ctypes.byref(num), ctypes.byref(buf), ctypes.byref(size)):
-                    return wszarray_to_list(buf)
-            return []
-
-        else:	# POSIX
+        elif platform != 'win32':
+            # POSIX
             lang = locale.getlocale()[0]
-            return lang and [lang.replace('_','-')] or []
+            return lang and [lang.replace('_', '-')] or []
+
+        def wszarray_to_list(array):
+            offset = 0
+            while offset < len(array):
+                sz = ctypes.wstring_at(ctypes.addressof(array) + offset*2)
+                if sz:
+                    yield sz
+                    offset += len(sz)+1
+
+                else:
+                    break
+
+        num = ctypes.c_ulong()
+        size = ctypes.c_ulong(0)
+        if GetUserPreferredUILanguages(MUI_LANGUAGE_NAME, ctypes.byref(num), None, ctypes.byref(size)) and size.value:
+            buf = ctypes.create_unicode_buffer(size.value)
+
+            if GetUserPreferredUILanguages(MUI_LANGUAGE_NAME, ctypes.byref(num), ctypes.byref(buf), ctypes.byref(size)):
+                return wszarray_to_list(buf)
+
+        return []
+
 
 # singletons
-Locale = Locale()
-Translations = Translations()
+Locale = _Locale()
+Translations = _Translations()
 
 
 # generate template strings file - like xgettext
```

Comment:
OH, that's what this code is doing.

This makes me nervous, because I doubt this is going to handle where we split the arguments to `_(...)` calls over multiple lines.  If anyone runs this in the future it will generate an incomplete file.

At the very least I think we need to change the template file to have a `.new` extension so you can compare it to the current one before deciding to use it or not.

Reply:
Yeah thats not a bad idea. Honestly using the python AST to find these is PROBABLY a better idea, if I can figure out how. This isnt exactly the smartest method to doing this.

Reply:
Okay Ive reimplemented this (see: https://gist.github.com/A-UNDERSCORE-D/8bb1d769c51295e827ec92951c464d94 ) And will add a new PR to replace this code once this PR is merged. In the interim, I've updated the old code to create the file with a .new as suggested.